source/agora/node/Registry.d

```diff
@@ -235,6 +231,14 @@ public class NameRegistry: NameRegistryAPI
         ensure(this.zones[ZoneIndex.Validator].type != ZoneType.caching,
             "Couldn't register, server is not authoritative for the zone");
 
+        ensure(registry_payload.verifySignature(registry_payload.data.public_key),
+               "Incorrect signature for payload");
+        this.registerValidator(registry_payload);
+    }
+
+    /// Similar to `postValidator`, but does not perform signature verification
+    public void registerValidator (RegistryPayload registry_payload) @safe
```

Comment:
We should be careful not to return the records that we obtained this way, as a response to an external request. I am not sure if that is possible, but just came to my mind.

Reply:
Hum, why ? If we're caching, anything goes. If we're authoritative, either this shouldn't get called (there's no upstream) or our upstream should be another authoritative server.

Reply:
I wasnt sure if that was possible in practice anyways.

> our upstream should be another authoritative server.

if they are to be trusted, I dont see a problem either